projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -158,18 +158,32 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   validate(): ValidationErrors {
-    const errors: ValidationErrors = {};
+    const invalidFields: Array<string> = [];
     Object.keys(this.sortCodeFormGroup.controls).forEach((fieldName) => {
       if (
         this.sortCodeFormGroup.controls[fieldName].invalid &&
         (this.sortCodeFormGroup.controls[fieldName].dirty ||
           (this.formGroupDirective && this.formGroupDirective.submitted))
       ) {
-        errors.invalidField = true;
+        invalidFields.push(fieldName);
       }
     });
-    if (Object.keys(errors).length) {
-      return errors;
+
+    const requiredFields = invalidFields.filter((fieldName) =>
+      this.sortCodeFormGroup.controls[fieldName].hasError('required'),
+    );
+
+    const error: ValidationErrors = {};
+    // If all the fields have been left empty we show the
+    // required error otherwise we show the invalid error.
+    if (requiredFields.length === 3) {
```

Comment:
I was skeptical of the magic number 3 here, and was wondering if there is a different way to identify that. Given we have a first, second and third input field I think it's actually ok.

Reply:
I could replace it with `Object.keys(this.sortCodeFormGroup.controls).length` but not sure that's better 🤔  
The sort code always have three controls and the above will make the code a bit less performant. What does everyone else think? happy to go with the majority :) 
@paul-request @owensgit

Reply:
Yeah, I think in then context of this component `3` is ok

Reply:
I'm happy with `3` in this case too 👍

Reply:
Cool thanks 👍